Mage/Module.cs

```diff
@@ -0,0 +1,74 @@
+﻿using System;
+using System.Collections.Generic;
+
+using Newtonsoft.Json.Linq;
+
+
+public class Module<T> : Singleton<T> where T : class, new() {
+	//
+	protected Mage mage { get { return Mage.instance; } }
+	protected Logger logger { get { return mage.logger(this.GetType().Name); } }
```

Comment:
Should this not be a singleton value? You're creating a new one every time `logger` is accessed, it seems.